src/backend/metal/src/native.rs

```diff
@@ -81,13 +82,30 @@ unsafe impl Sync for ComputePipeline {}
 #[derive(Debug)]
 pub struct Image {
     pub(crate) raw: metal::Texture,
-    pub(crate) extent: hal::image::Extent,
+    pub(crate) allocations: Option<Arc<Mutex<MemoryAllocations>>>,
+    pub(crate) extent: image::Extent,
+    pub(crate) num_layers: Option<image::Layer>,
     pub(crate) format_desc: hal::format::FormatDesc,
     pub(crate) shader_channel: Channel,
     pub(crate) mtl_format: metal::MTLPixelFormat,
     pub(crate) mtl_type: metal::MTLTextureType,
 }
 
+impl Image {
+    pub(crate) fn pitches_impl(
+        extent: image::Extent, format_desc: hal::format::FormatDesc
+    ) -> [hal::buffer::Offset; 3] {
+        let bytes_per_texel = format_desc.bits as image::Size >> 3;
+        let row_pitch = extent.width * bytes_per_texel;
+        let depth_pitch = extent.height * row_pitch;
+        let array_pitch = extent.depth * depth_pitch;
+        [row_pitch as _, depth_pitch as _, array_pitch as _]
```

Comment:
We could just use a tuple here so we can destructure with `let (.., array_pitch)` in the callers (instead of `pitches[2]`)

Reply:
we could, yeah